### PR TITLE
Perform max, min, average, standard deviation, median, and mode calculations

### PR DESCRIPTION
**Pull request by will-hou**

### CODE REVIEW

File: src/app/data/calculations.ts [hunk 1]
```diff
@@ -13,18 +14,30 @@ export interface Calculation<T = any> {
 export const calculations: Calculations = {
     'Maximum': {
         type: ScoutMetricType.NUMBER,
-        invoke: (...metrics: Array<Metric>) => metrics[0]
+        invoke: (...metrics: Array<Metric>) => reduce(metrics, function (acc: Metric, metric) {
```
Comment (andrewda) [inline src/app/data/calculations.ts]: Not too big of a deal, but arrow functions are beautiful

```suggestion
        invoke: (...metrics: Array<Metric>) => reduce(metrics, (acc: Metric, metric) => {
```

File: src/app/data/calculations.ts [hunk 2]
```diff
@@ -13,18 +14,30 @@ export interface Calculation<T = any> {
...
+            if (metric.value > acc.value) acc.value = metric.value
+            return acc
+        }, { type: ScoutMetricType.NUMBER, value: 0, category: metrics[0].category})
     },
     'Minimum': {
         type: ScoutMetricType.NUMBER,
-        invoke: (...metrics: Array<Metric>) => metrics[0]
+        invoke: (...metrics: Array<Metric>) => reduce(metrics, function(acc: Metric, metric) {
```
Comment (andrewda) [inline src/app/data/calculations.ts]:
```suggestion
        invoke: (...metrics: Array<Metric>) => reduce(metrics, (acc: Metric, metric) => {
```

File: src/app/data/calculations.ts [hunk 3]
```diff
@@ -13,18 +14,30 @@ export interface Calculation<T = any> {
...
+            if (metric.value > -1 && acc.value == -1) acc.value = metric.value
+            else if (metric.value < acc.value) acc.value = metric.value
+            return acc
+          }, {type: ScoutMetricType.NUMBER, value: -1, category: metrics[0].category})
     },
     'Average': {
         type: ScoutMetricType.NUMBER,
-        invoke: (...metrics: Array<Metric>) => metrics[0]
+        invoke: (...metrics: Array<Metric>) => ({type:ScoutMetricType.NUMBER, value: round(reduce(metrics, function (acc, metric){
```
Comment (andrewda) [inline src/app/data/calculations.ts]:
```suggestion
        invoke: (...metrics: Array<Metric>) => ({type:ScoutMetricType.NUMBER, value: round(reduce(metrics, (acc, metric) => { 
```

File: src/app/data/calculations.ts [hunk 4]
```diff
@@ -13,18 +14,30 @@ export interface Calculation<T = any> {
...
         type: ScoutMetricType.NUMBER,
-        invoke: (...metrics: Array<Metric>) => metrics[0]
+        invoke: (...metrics: Array<Metric>) => ({type:ScoutMetricType.NUMBER, value: round(reduce(metrics, function (acc, metric){ 
+            return acc + metric.value}, 0) / metrics.length, 2), category: metrics[0].category
+        })
     },
     'Standard Deviation': {
         type: ScoutMetricType.NUMBER,
-        invoke: (...metrics: Array<Metric>) => metrics[0]
+        invoke: (...metrics: Array<Metric>) => ({type: ScoutMetricType.NUMBER, value: round(std(reduce(metrics, function(acc, metric) {
```
Comment (andrewda) [inline src/app/data/calculations.ts]:
```suggestion
        invoke: (...metrics: Array<Metric>) => ({type: ScoutMetricType.NUMBER, value: round(std(reduce(metrics, (acc, metric) => { 
```

File: src/app/data/calculations.ts [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import { Metric, ScoutMetricType } from '@robot-analytics/data/metric';
-import { reduce, map } from 'lodash';
+import { reduce } from 'lodash';
+import { std, round } from 'mathjs'
```
Comment (andrewda) [inline src/app/data/calculations.ts]: Missing semicolon (semicolon rules aren't strictly enforced -- we should get linting set up soon!)

```suggestion
import { std, round } from 'mathjs';
```

**Comment by andrewda** [comment]
Not sure this should be in this PR or another, but having mode and median calculations would be cool too in order to deal with outliers.

**Comment by will-hou** [comment]
Good idea! Hehe I can edit the name of the pull request and make it applicable 😉

**Pull request merged**